src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py

```diff
@@ -677,6 +677,9 @@ def __init__(self, config, add_pooling_layer=True):
 
         self.pooler = XLMRobertaXLPooler(config) if add_pooling_layer else None
 
+        self._use_sdpa = config._attn_implementation == "sdpa"
+        self.position_embedding_type = config.position_embedding_type
```

Comment:
We shouldn't set this here - as no error will be thrown as in the other models when trying to instantiate the model. This model could have `self._use_sdpa` set to True, even if it's not using SDPA, which can cause all sorts of errors and confusion.